src/main/scala/uk/gov/hmrc/ct/ct600/v3/calculations/B325Calculator.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ct.ct600.v3.calculations
+
+import uk.gov.hmrc.ct.box.CtTypeConverters
+import uk.gov.hmrc.ct.computations.CP291
+import uk.gov.hmrc.ct.ct600.v3._
+
+trait B325Calculator extends CtTypeConverters {
+
+  def calculateB325(b335: B335, b385: B385, b315: B315, b330: B330, b380: B380, cp291: CP291): B325 = {
+
+    val ZERO = Some(0)
+
+    val value =
+      if (!b330.isPositive) None
+      else b380 match {
+        case noTradingProfit if cp291.noValue => ZERO
+        case singleFinancialYear if singleFinancialYear.noValue =>
+          if (b335.value <= b315.value) Some(b335.value)
+          else ZERO
+        case twoFinancialYears if twoFinancialYears.hasValue =>
+          if ((b335.value + b385.value) <= b315.value) Some(b335.value + b385.value)
+          else ZERO
```

Comment:
Check it now. Zero is more often here. @rutpatelhmrc @AlokHMRC

Reply:
Still I don't see any condition of NIR eligibility.
Not sure how it should look like properly.

Reply:
This just calculates the figure for B325. The condition of NIR eligibility has been done in filing.